jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp

```diff
@@ -1667,6 +1667,127 @@ ecma_builtin_string_prototype_object_slice (ecma_value_t this_arg, /**< this arg
   return ret_value;
 } /* ecma_builtin_string_prototype_object_slice */
 
+#ifndef CONFIG_ECMA_COMPACT_PROFILE_DISABLE_REGEXP_BUILTIN
+
+/**
+ * The abstract SplitMatch routine for String.prototype.split()
+ *
+ * See also:
+ *          ECMA-262 v5, 15.5.4.14
+ *
+ * Used by:
+ *        - The String.prototype.split routine.
+ *
+ * @return completion value - contains the value of the match
+ *                          - the index property of the completion value indicates the position of the
+ *                            first character in the input_string that matched
+ *
+ *         Returned value must be freed with ecma_free_completion_value.
+ */
+static ecma_completion_value_t
+ecma_builtin_helper_split_match (ecma_value_t input_string, /**< first argument */
+                                 ecma_length_t start_idx, /**< second argument */
+                                 ecma_value_t separator) /**< third argument */
+{
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+
+  /* 1. */
+  if (ecma_is_value_object (separator)
+      && ecma_object_get_class_name (ecma_get_object_from_value (separator)) == LIT_MAGIC_STRING_REGEXP_UL)
+  {
+    ecma_value_t regexp_value = ecma_copy_value (separator, false);
+
+    ECMA_TRY_CATCH (to_string_val,
+                    ecma_op_to_string (input_string),
+                    ret_value);
+
+    ecma_string_t *input_str_p = ecma_get_string_from_value (to_string_val);
+    ecma_string_t *substr_str_p = ecma_string_substr (input_str_p, start_idx, ecma_string_get_length (input_str_p));
+
+    ret_value = ecma_regexp_exec_helper (regexp_value, ecma_make_string_value (substr_str_p), true);
+
+    if (!ecma_is_value_null (ecma_get_completion_value_value (ret_value)))
+    {
+      ecma_object_t *obj_p = ecma_get_object_from_completion_value (ret_value);
+      ecma_string_t *magic_str_p = ecma_get_magic_string (LIT_MAGIC_STRING_INDEX);
+      ecma_property_t *index_prop_p = ecma_find_named_property (obj_p, magic_str_p);
+
+      JERRY_ASSERT (index_prop_p != NULL);
```

Comment:
`ecma_get_named_property` could be used in the case. The version includes assertion from line 1715.